awscli/customizations/cloudformation/deployer.py

```diff
@@ -106,6 +108,17 @@ def create_changeset(self, stack_name, cfn_template,
             'Capabilities': capabilities,
             'Description': description,
         }
+
+        # If an S3 uploader is available, use TemplateURL to deploy rather than
+        # TemplateBody. This is required for large templates.
+        if s3_uploader:
+            with mktempfile() as temporary_file:
+                temporary_file.write(kwargs.pop('TemplateBody'))
+                temporary_file.flush()
+                url = s3_uploader.upload_with_dedup(
+                        temporary_file.name, "template")
+                kwargs['TemplateURL'] = s3_uploader.to_path_style_s3_url(url.split('/')[-1])
```

Comment:
This URL creation should be changing based on the parameter _s3_prefix_. If not, the _CreateChangeSet_ step would get a _object does not exist_ error.

Reply:
Example error:

```
examplePrefix/examplehash.template  100000 / 100000  (100.00%)
An error occurred (ValidationError) when calling the CreateChangeSet operation: Template file referenced by https://s3.amazonaws.com/examplebucketname/examplehash.template does not exist.
```

Reply:
This seems to be a bug  in the _s3_uploader.to_path_style_s3_url_ method. Ideally, the prefix part of the result url should be:
```
base = "{0}/{1}/{2}".format(base, self.bucket_name, self.prefix)
```
when _self.prefix_ is present.

Reply:
This is not a bug in to_path_style_s3_url, but a bug in the url.split. Parsing a S3 URL has more nuances. Correct implementation should follow this: https://github.com/aws/aws-cli/blob/develop/awscli/customizations/cloudformation/artifact_exporter.py#L351-L357